### ISSUE TITLE
Other types of coronagraphs

### ISSUE THREAD

**Issue by udik-chudik** [ajeldorado/falco-matlab#183]
Hi guys! I see your optimizer currently supports some kinds of Liot coronagraph, what about other types? With such poor documentation I cant figure out program`s architecture. Could you describe the key concepts/starting point to let me add support rotational shearing interferometric coronagraph?

**Comment by ajeldorado** [comment]
Could you provide a link to a paper describing the rotational shearing interferometric coronagraph? I'm not familiar with it, so I'd have to look to see if it would work with the algorithms included in FALCO.

**Comment by udik-chudik** [comment]
Yes of course, look at [this](https://www.spiedigitallibrary.org/journals/Journal-of-Astronomical-Telescopes-Instruments-and-Systems/volume-4/issue-4/044001/Stellar-imaging-coronagraph-and-exoplanet-coronal-spectrometer--two-additional/10.1117/1.JATIS.4.4.044001.short) for example.
Basically, from model point of view, it is two copies of pupil plane, rotated at specified angle, that sums together with PI phase shift.

**Comment by udik-chudik** [comment]
Hi! Is there any updates?

**Comment by ajeldorado** [comment]
I won't have time to add the RSI option in, at least not soon. Would the rotation of the pupil always be an easy value such as +/-90 or +/-180 degrees? If not, then the Jacobian calculation would get trickier (and take more time to code) since it would have to rotate a subwindow of the pupil plane array and propagate that in addition to the unchanged half of the recombined beam.

**Comment by udik-chudik** [comment]
I could try to add the RSI option by my own and make a pull request (I have enough time to do that :relaxed:), but I need some general information about FALCO program structure, common functions and optimization path, maybe some advice to get start with and what to look at. I am familiar with EFC principals, the question is in implementation.
As for rotation angle: in general there could be any arbitrary angle. But I think hard-coded 180 deg would be enough to begin with.
